boa/src/builtins/value/display.rs

```diff
@@ -67,6 +67,14 @@ pub(crate) fn log_string_from(x: &Value, print_internals: bool) -> String {
             match v.borrow().data {
                 ObjectData::String(ref string) => format!("String {{ \"{}\" }}", string),
                 ObjectData::Boolean(boolean) => format!("Boolean {{ {} }}", boolean),
+                ObjectData::Number(rational) => {
```

Comment:
Why doesn't this use `format_rational() to reduce code duplication?

Reply:
> Why doesn't this use `format_rational() to reduce code duplication?

Yes. I wanted to use `format_rational` but it takes a `std::Formatter` and the other `log_*` functions don't maybe I should change them too so they take a `std::Formatter`

Reply:
> Yes. I wanted to use `format_rational` but it takes a `std::Formatter` and the other `log_*` functions don't maybe I should change them too so they take a `std::Formatter`

I think I'll leave it as is for now, the way things are displayed is a mess, and trying to change it to use `std::Formatter` would require an almost complete rewrite of it and I think it would be best to do this in a separate PR.